src/mmrelay/matrix_utils.py

```
@@ -2955,125 +3091,163 @@ async def on_room_message(
+        )
+        return response
+    except NIO_COMM_EXCEPTIONS as e:
+        # Convert nio communication exceptions to ImageUploadError
+        # Create a mock UploadError for the ImageUploadError constructor
+        mock_upload_error = UploadError(message=str(e))
+        raise ImageUploadError(mock_upload_error) from e
+    except asyncio.TimeoutError as e:
+        # Convert timeout exceptions to ImageUploadError
+        mock_upload_error = UploadError(message=f"Upload timeout: {e}")
+        raise ImageUploadError(mock_upload_error) from e
+    except asyncio.TimeoutError as e:
```

Ruff (0.14.5)

try-except block with duplicate exception `asyncio.TimeoutError`

(B025)